### PR TITLE
feat: modern build

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [x] New feature (a non-breaking change which adds functionality)

## Description

- [x] SPA modern build
- [x] SSR modern rendering
- [x] CLI command
- [x] Test
- [ ] Documentation
- [ ] Release Note

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/webpack/src/config/plugins/vue/modern.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        const tempFilename = path.join(this.targetDir, htmlPath, `legacy-assets-${htmlName}.json`)
+        await fs.mkdirp(path.dirname(tempFilename))
+        await fs.writeFile(tempFilename, JSON.stringify(data.body))
+        cb()
+      })
+    })
+  }
+
+  applyModern(compiler) {
+    const ID = `vue-cli-modern-bundle`
```
Comment (pi0) [inline packages/webpack/src/config/plugins/vue/modern.js]: vue-cli?

Comment (clarkdo) [reply packages/webpack/src/config/plugins/vue/modern.js]: The plugin is from vue-cli, I’m wondering what we should do, maybe changing it and prepending credit comment?

Comment (pi0) [reply packages/webpack/src/config/plugins/vue/modern.js]: I think keeping credit/license in the code should be enough. You know it is somehow unusual to see a `vue-cli` related error when debugging a Nuxt app 😆

File: packages/webpack/src/config/modern.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import _ from 'lodash'
```
Comment (manniL) [inline packages/webpack/src/config/modern.js]: Please use single imports for lodash :raised_hands:

Comment (clarkdo) [reply packages/webpack/src/config/modern.js]: Good catch

File: packages/vue-renderer/src/renderer.js [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
     // Will be set by createRenderer
-    this.bundleRenderer = null
-    this.spaMetaRenderer = null
+    this.renderer = {
+      ssr: null,
+      modern: null,
+      spa: null
+    }
+
+    this.modernBrowsers = Object.keys(ModernBrowsers)
```
Comment (pi0) [inline packages/vue-renderer/src/renderer.js]: Can't we precompute this and put what we cant into JSON file? (To reduce startup time)

File: packages/webpack/src/config/base.js [hunk 4]
```diff
@@ -25,11 +26,20 @@ export default class WebpackBaseConfig {
     this.loaders = this.options.build.loaders
   }
 
+  get colors() {
+    return {
+      client: 'green',
+      server: 'orange',
+      modern: 'blue'
```
Comment (pi0) [inline packages/webpack/src/config/base.js]: cyan look better i think :D

**Review by pi0 — APPROVED** [review]

**Pull request merged**